app/invocation/invocation_cancel_button.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Slash as SlashIcon } from "lucide-react";
+import { OutlinedButton } from "../components/button/button";
+import rpcService from "../service/rpc_service";
+import { invocation } from "../../proto/invocation_ts_proto";
+
+export interface InvocationCancelButtonComponentProps {
+  invocationId: string;
+}
+
+export default class InvocationCancelButtonComponent extends React.Component<InvocationCancelButtonComponentProps> {
+  private onClick() {
+    rpcService.service.cancelInvocation(
+      new invocation.CancelInvocationRequest({ invocationId: this.props.invocationId })
+    );
+  }
+
+  render() {
+    return (
+      <div className="invocation-cancel-button-container">
+        <OutlinedButton onClick={this.onClick.bind(this)}>
+          <SlashIcon className="icon" />
+          <div>Cancel</div>
+        </OutlinedButton>
+      </div>
+    );
```

Comment:
nit: add a loading state, and also set the `disabled` prop on the button while the RPC is in progress, so that the user can't fire duplicate RPCs, and also so they can get visual feedback as to whether the RPC has completed or not.

You can copy some of the logic from `workflow_rerun_button` for that, mainly the `isLoading` state logic in the function that makes the RPC, and also showing the `<Spinner />` component instead of the SlashIcon while it's loading

to test the loading state locally, you can set the network speed in chrome devtools to something like Fast 3G so that the local RPC is slower, since usually they are almost instant. (I'd recommend only slowing down the network after the app is loaded, since it takes forever to load the full app on Fast 3G settings)

![image](https://user-images.githubusercontent.com/2414826/179599456-3443c459-944c-48db-a902-e394f38e6e8f.png)

Reply:
Done